Readme.md

```diff
@@ -1,5 +1,154 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set some rules
+
+First you will need to add some rules to the BNF before we can set the rules
+```rust
+NumberList ::= number (sep number)*
+sep ::= " "
+number ::= [0-9]+
+```
+`add_rule_for_names` takes in a 2 arguments. The First one being a `list` or in our case `vec!` of names you want to set a rule for.
```

Comment:
`add_rule_for_names` is not qualified by any type, so the reader doesn't automatically know what you're referring to. It belongs to the `Lexer` type, and it'd be easiest to just show it being invoked on a lexer as an example.

`vec!` is the name of the macro used to create a `Vec` type shorthand. Rather than giving a description of the function to start with, it might be helpful to give an example of token output for a simple input using the above BNF. That will give the reader an understanding of what the BNF is doing and also be a good way to introduce the problem of excess tokens that aren't needed for parsing, leading into explaining culling rules. Trying to explain what this does in a vacuum is likely to just leave people confused.